pilot/pkg/networking/core/v1alpha3/peer_authentication_simulation_test.go

```diff
@@ -0,0 +1,614 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha3_test
+
+import (
+	"testing"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/simulation"
+	"istio.io/istio/pilot/pkg/xds"
+)
+
+// TestPeerAuthenticationPassthrough tests the PeerAuthentication policy applies correctly on the passthrough filter chain,
+// including both global configuration and port level configuration.
+func TestPeerAuthenticationPassthrough(t *testing.T) {
+	paStrict := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+ name: default
+spec:
+ selector:
+   matchLabels:
+     app: foo
+ mtls:
+   mode: STRICT
+---`
+	paDisable := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+ name: default
+spec:
+ selector:
+   matchLabels:
+     app: foo
+ mtls:
+   mode: DISABLE
+---`
+	paPermissive := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+ name: default
+spec:
+ selector:
+   matchLabels:
+     app: foo
+ mtls:
+   mode: PERMISSIVE
+---`
+	paStrictWithDisableOnPort9000 := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+ name: default
+spec:
+ selector:
+   matchLabels:
+     app: foo
+ mtls:
+   mode: STRICT
+ portLevelMtls:
+   9000:
+     mode: DISABLE
+---`
+	paDisableWithStrictOnPort9000 := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+ name: default
+spec:
+ selector:
+   matchLabels:
+     app: foo
+ mtls:
+   mode: DISABLE
+ portLevelMtls:
+   9000:
+     mode: STRICT
+---`
+	paDisableWithPermissiveOnPort9000 := `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+spec:
+  selector:
+    matchLabels:
+      app: foo
+  mtls:
+    mode: DISABLE
+  portLevelMtls:
+    9000:
+      mode: PERMISSIVE
+---`
+	sePort8000 := `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+ name: se
+spec:
+ hosts:
+ - foo.bar
+ endpoints:
+ - address: 1.1.1.1
+ location: MESH_INTERNAL
+ resolution: STATIC
+ ports:
+ - name: http
+   number: 8000
+   protocol: HTTP
+---`
+	mkCall := func(port int, tls simulation.TLSMode) simulation.Call {
+		// TODO https://github.com/istio/istio/issues/28506 address should not be required here
+		r := simulation.Call{Port: port, CallMode: simulation.CallModeInbound, TLS: tls, Address: "1.1.1.1"}
+		if tls == simulation.MTLS {
+			r.Alpn = "istio"
+		}
+		return r
+	}
+	cases := []struct {
+		name   string
+		config string
+		calls  []simulation.Expect
+	}{
+		{
+			name:   "global disable",
+			config: paDisable,
+			calls: []simulation.Expect{
+				{
+					Name:   "mtls",
+					Call:   mkCall(8000, simulation.MTLS),
+					Result: simulation.Result{Error: simulation.ErrNoFilterChain},
```

Comment:
I think this is not matching envoy, per my comment on the simulator logic change. Try it in a real cluster:
```
{"destinationPort":15006}
{"prefixRanges":[{"addressPrefix":"0.0.0.0","prefixLen":0}],"transportProtocol":"raw_buffer"}
{"prefixRanges":[{"addressPrefix":"0.0.0.0","prefixLen":0}],"transportProtocol":"raw_buffer","applicationProtocols":["http/1.0","http/1.1","h2c"]}
{"destinationPort":7070,"transportProtocol":"raw_buffer"}
{"destinationPort":7071,"transportProtocol":"raw_buffer"}
{"destinationPort":8080,"transportProtocol":"raw_buffer"}
{"destinationPort":8443,"transportProtocol":"raw_buffer"}
{"destinationPort":9090,"transportProtocol":"raw_buffer"}
```

filter chains

then `$ curl https://echo:8443 -k` will send TLS traffic that reaches the application pod

Reply:
okay, reverted this change.